awx/sso/fields.py

```diff
@@ -335,19 +370,66 @@ def __init__(self, choices=None, **kwargs):
 
     def to_representation(self, value):
         if not value:
-            return ''
+            return 'MemberDNGroupType'
         if not isinstance(value, django_auth_ldap.config.LDAPGroupType):
             self.fail('type_error', input_type=type(value))
         return value.__class__.__name__
 
     def to_internal_value(self, data):
+        def find_class_in_modules(class_name):
+            module_search_space = [django_auth_ldap.config, awx.sso.ldap_group_types]
+            for m in module_search_space:
+                cls = getattr(m, class_name, None)
+                if cls:
+                    return cls
+            return None
+
         data = super(LDAPGroupTypeField, self).to_internal_value(data)
         if not data:
             return None
+
+        params = self.get_depends_on() or {}
+        cls = find_class_in_modules(data)
+        if not cls:
+            return None
+
+        # Per-group type parameter validation and handling here
+
+        # Backwords compatability. Before AUTH_LDAP_GROUP_TYPE_PARAMS existed
+        # MemberDNGroupType was the only group type, of the underlying lib, that
+        # took a parameter.
+        params_sanitized = dict()
+        if isinstance(cls, LDAPGroupType):
+            for k in VALID_GROUP_TYPE_PARAMS_MAP['LDAPGroupType']:
```

Comment:
I would think all of this could be simplified by using `inspect.getargspec(cls.__init__).args[1:]` to determine which arguments are supported for the given class.

Reply:
Was worried about kwargs.

Reply:
None of the built-in group types accept unspecified kwargs. If you modified `PosixUIDGroupType` to only accept `name_attr` and `ldap_group_user_attr`, then it would work.